esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6754,4 +6754,149 @@ void CWsDfuEx::setPublishFileSize(const char *lfn, IFileDescriptor *fileDesc)
     async.For(fileDesc->numParts(), 100);
 }
 
+struct _ownerInfo {
+    std::string name;
+    long long int totalFiles;
+    long long int totalSize;
+};
+
+struct _groupInfo {
+    std::string name;
+    long long int totalSize;
+    long long int totalFiles;
+    std::map<std::string, std::shared_ptr<_ownerInfo>> owners;
```

Comment:
I very much doubt the overhead is significant in either case. We should be worrying more about maintainability/obvious correctness for most code...

Reply:
agree. unique_ptr makes more sense here from a readability standpoint, since the map is the sole owner, there is no sharing or link counting required.